components/builder-worker/src/runner/mod.rs

```diff
@@ -0,0 +1,350 @@
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to spawn child");
+        if let Some(ref mut stdout) = child.stdout {
+            for line in BufReader::new(stdout).lines() {
+                let mut l: String = line.unwrap();
+                let eol: &str = "\n";
+                l = l + eol;
+                logger.log(l.as_bytes());
+            }
+        }
+        let exit_status = child.wait().expect("failed to wait on child");
+        debug!("build complete, status={:?}", exit_status);
+        if exit_status.success() {
+            try!(fs::rename(self.workspace.src().join("results"), self.workspace.out()));
+            self.workspace.last_built()
+        } else {
+            Err(Error::BuildFailure(exit_status.code().unwrap_or(-1)))
+        }
+    }
+
+    pub fn job(&self) -> &proto::Job {
+        &self.workspace.job
+    }
+
+    fn complete(&mut self) -> () {
+        self.teardown().err().map(|e| error!("{}", e));
+        self.workspace.job.set_state(proto::JobState::Complete);
+    }
+
+    fn fail(&mut self) -> () {
+        self.teardown().err().map(|e| error!("{}", e));
+        self.workspace.job.set_state(proto::JobState::Failed);
+    }
+
+    fn post_process(&self, archive: &mut PackageArchive) -> bool {
+        // JW TODO: In the future we'll support multiple and configurable post processors, but for
+        // now let's just publish to the public depot
+        //
+        // Things to solve right now
+        // * Where do we get the token for authentication?
+        //      * Should the workers ask for a lease from the JobSrv?
+        let client =
+            depot_client::Client::new(&hab_core::url::default_depot_url(), PRODUCT, VERSION, None)
+                .unwrap();
+        if let Some(err) = client.x_put_package(archive, &self.auth_token).err() {
+            error!("post processing error, ERR={:?}", err);
+            return false;
+        }
+        if let Some(err) = fs::remove_dir_all(self.workspace.out()).err() {
+            error!("unable to remove out directory ({}), ERR={:?}",
+                   self.workspace.out().display(),
+                   err)
+        }
+        true
+    }
+
+    fn setup(&self) -> Result<()> {
+        if let Some(err) = fs::create_dir_all(self.workspace.src()).err() {
+            return Err(Error::WorkspaceSetup(format!("{}", self.workspace.src().display()), err));
+        }
+        Ok(())
+    }
+
+    fn teardown(&self) -> Result<()> {
+        let args =
+            vec![OsString::from("-s"),
+                 OsString::from(self.workspace.src()),
+                 OsString::from("-r"),
+                 OsString::from(self.workspace.studio()),
+                 OsString::from("rm"),
+                 OsString::from(Path::new(self.workspace.job.get_project().get_plan_path())
+                     .parent()
+                     .unwrap())];
+        let command = match PackageInstall::load(&STUDIO_PKG, None) {
```

Comment:
We've got this install hab studio with the same panic in a couple of spots (here in `teardown` and one in `build`), maybe extract to a common function?  Especially given the earlier comment that we'll need to add some additional logic around determining the version of the studio package?